src/Fl_Window_Driver.cxx

```diff
@@ -286,6 +286,34 @@ Fl_Window *Fl_Window_Driver::find(fl_uintptr_t xid) {
   return 0;
 }
 
+
+void Fl_Window_Driver::maximize() {
+  *no_fullscreen_x() = x();
+  *no_fullscreen_y() = y();
+  *no_fullscreen_w() = w();
+  *no_fullscreen_h() = h();
+  int X,Y,W,H;
```

Comment:
Member functions `no_fullscreen_{x,y,w,h}()` are accessors to variables `no_fullscreen_{x,y,w,h}` which are shared with the fullscreen-handling mechanism which uses 0 (not -1) as reset indicator.

The new code introduces `Fl_Window::maximize_active()` which is where FLTK checks whether a window is maximized. If so, it will do nothing on a second `Fl_Window::maximize()` call.